source/server/http/admin.cc

```diff
@@ -497,6 +511,38 @@ std::string AdminImpl::statsAsJson(const std::map<std::string, uint64_t>& all_st
     stat_obj.AddMember("value", stat_value, allocator);
     stats_array.PushBack(stat_obj, allocator);
   }
+
+  for (Stats::ParentHistogramSharedPtr histogram : all_histograms) {
+    Value histogram_obj;
+    histogram_obj.SetObject();
+    Value histogram_name;
+    histogram_name.SetString(histogram->name().c_str(), allocator);
+    histogram_obj.AddMember("name", histogram_name, allocator);
+
+    rapidjson::Value quantile_array(rapidjson::kArrayType);
+
+    for (size_t i = 0; i < histogram->intervalStatistics().supportedQuantiles().size(); ++i) {
```

Comment:
Yeah to Matt's point elsewhere: supportedQuantiles() is common across all histograms, and thus should be static const (though this requires some care to avoid statically constructing a std::vector which violates style-guide).  This leads me also to this observation, which I'm fine either way with.

What you have is fine from a data-model perspective and is readable, but if we care about compactness you can make it smaller. You could show the supportedQuantiles array once, and the histograms can be rendered as two arrays (intervals and cumulative).  This might matter if we are streaming a lot of stats, or there are a lot of histograms, and we are streaming over a socket to a remote host.

Reply:
Yeah +1 I think we can optimize. Right now since we don't have locked protos (or any protos) for the JSON output, I think we can change this stuff at any time. I would put in a TODO to think about optimizing when we switch over to protos for all admin JSON output.

Reply:
right a TODO should be here -- actually a warning not to use the data until another round of optimization.